ledger/participant-integration-api/src/main/scala/platform/store/backend/m/MEventStorageBackend.scala

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.m
+
+import java.sql.Connection
+
+import com.daml.ledger.offset.Offset
+import com.daml.lf.data.{Ref, Time}
+import com.daml.lf.data.Ref.{Party, TransactionId}
+import com.daml.lf.ledger.EventId
+import com.daml.logging.LoggingContext
+import com.daml.platform.store.appendonlydao.events.{ContractId, EventsTable, Raw}
+import com.daml.platform.store.backend.EventStorageBackend.FilterParams
+import com.daml.platform.store.backend.{DbDto, EventStorageBackend}
+import com.daml.platform.store.cache.LedgerEndCache
+
+import scala.collection.immutable.ArraySeq
+import scala.collection.Searching.{Found, InsertionPoint}
+
+import DbDtoUtils._
+
+class MEventStorageBackend(ledgerEndCache: LedgerEndCache) extends EventStorageBackend {
+
+  override def pruneEvents(
+      pruneUpToInclusive: Offset,
+      pruneAllDivulgedContracts: Boolean,
+  )(connection: Connection, loggingContext: LoggingContext): Unit = MStore.update(connection) {
+    mStore =>
+      val pruneUpToInclusiveString = pruneUpToInclusive.toHexString.toString
+      mStore.copy(
+        contractIdIndex = mStore.contractIdIndex.iterator.flatMap { case (contractId, dtos) =>
```

Comment:
pruning is implemented only for observe-ability: no real pruning happens to events, as event_sequential_id and the index in the event store match.